base_domain_operator/expression.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import api
+# pylint: disable=W0402
+from openerp.osv.expression import ExtendedLeaf, expression, is_operator,\
+    is_leaf
+
+
+class BaseDomainOperatorExtendedLeaf(ExtendedLeaf):
+    def __init__(self, leaf, model, join_context=None, internal=False):
+        if isinstance(leaf, ExtendedLeaf):
+            # TODO: think very long and hard if this introduces a security
+            # problem
+            internal = True
```

Comment:
I think being able to use `inselect` for the definition of other operators will be very helpful. The comment is meant to notify reviewers to look closely at this because of the security implications. To the best of my knowledge, this is not a problem...